### ISSUE TITLE
exception using isolate spawn in pub-build

### ISSUE THREAD

**Issue by DartBot** [dart-lang/pub#672]
<a href="https://github.com/sigmundch"><img src="https://avatars.githubusercontent.com/u/2049220?v=3" align="left" width="96" height="96"hspace="10"></img></a> **Issue by [sigmundch](https://github.com/sigmundch)**
_Originally opened as dart-lang/sdk#14201_

----

Try adding a transformer that is loaded dynamically in an isolate. When running this from the SDK's pub, I get the following error:

IsolateSpawnException: 'Unable to open file: /mnt/data/b/build/slave/dart-editor-linux-trunk/build/dart/sdk/lib/_internal/pub/bin/pub.dart'

I'm not sure if this is because pub is snapshoted?

If I run with '--trace', I get these additional details (I truncated the log transcript):

IsolateSpawnException: 'Unable to open file: /mnt/data/b/build/slave/dart-editor-linux-trunk/build/dart/sdk/lib/_internal/pub/bin/pub.dart'
dart:isolate                                                                                                                                                 spawnFunction
../../../../../../../../../../../mnt/data/b/build/slave/dart-editor-linux-trunk/build/dart/sdk/lib/_internal/pub/lib/src/dart.dart                           runInIsolate.&lt;fn&gt;
../../../../../../../../../../../mnt/data/b/build/slave/dart-editor-linux-trunk/build/dart/sdk/lib/_internal/pub/lib/src/io.dart                             withTempDir.&lt;fn&gt;.&lt;fn&gt;
dart:async                                                                                                                                                   Future.Future.sync
../../../../../../../../../../../mnt/data/b/build/slave/dart-editor-linux-trunk/build/dart/sdk/lib/_internal/pub/lib/src/io.dart                             withTempDir.&lt;fn&gt;
dart:async                                                                                                                                                   Future.Future.sync
../../../../../../../../../../../mnt/data/b/build/slave/dart-editor-linux-trunk/build/dart/sdk/lib/_internal/pub/lib/src/io.dart                             withTempDir
../../../../../../../../../../../mnt/data/b/build/slave/dart-editor-linux-trunk/build/dart/sdk/lib/_internal/pub/lib/src/dart.dart                           runInIsolate
../../../../../../../../../../../mnt/data/b/build/slave/dart-editor-linux-trunk/build/dart/sdk/lib/_internal/pub/lib/src/barback/load_transformers.dart      loadTransformers
../../../../../../../../../../../mnt/data/b/build/slave/dart-editor-linux-trunk/build/dart/sdk/lib/_internal/pub/lib/src/barback/load_all_transformers.dart  load
dart:async                                                                                                                                                   Future.wait
../../../../../../../../../../../mnt/data/b/build/slave/dart-editor-linux-trunk/build/dart/sdk/lib/_internal/pub/lib/src/barback/load_all_transformers.dart  loadAllTransformers.loadPackage.&lt;fn&gt;.&lt;fn&gt;
dart:async                                                                                                                                                   \_createTimer.&lt;fn&gt;
timer_impl.dart                                                                                                                                              \_Timer._createTimerHandler._handleTimeout
timer_impl.dart                                                                                                                                              \_Timer._createTimerHandler.&lt;fn&gt;
dart:isolate                                                                                                                                                 \_ReceivePortImpl._handleMessage

**Comment by DartBot** [comment]
<a href="https://github.com/sigmundch"><img src="https://avatars.githubusercontent.com/u/2049220?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [sigmundch](https://github.com/sigmundch)**

----

It is a bit strange though that the paths go up to the root and then down :-/

**Comment by DartBot** [comment]
<a href="https://github.com/nex3"><img src="https://avatars.githubusercontent.com/u/188?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [nex3](https://github.com/nex3)**

----

*Marked this as being blocked by dart-lang/sdk#14236.*

**Comment by DartBot** [comment]
<a href="https://github.com/nex3"><img src="https://avatars.githubusercontent.com/u/188?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [nex3](https://github.com/nex3)**

----

Issue dart-lang/pub#656 has been merged into this issue.
______
 cc @sigmundch.
 cc @jmesserly.

**Comment by DartBot** [comment]
<a href="https://github.com/nex3"><img src="https://avatars.githubusercontent.com/u/188?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [nex3](https://github.com/nex3)**

----

*Added [Fixed](../labels/Fixed) label.*
